### ISSUE TITLE
Angular 5 with angular cli - 'highcharts-chart' is not a known element

### ISSUE THREAD

**Issue by ChrystalLangston** [highcharts/highcharts-angular#22]
Hey there,

I am having the worst time trying to get my Angular 5 app to find the HighchartsChartComponent. I've outlined the steps that I followed.  Can you please tell me what I am doing wrong or maybe what I am forgetting?  Any help will be greatly appreciated!

Thank you in advance,
C

Here's the steps I took:
1. npm install highcharts/highcharts-angular
2. npm install highcharts --save  ... below is the resulting package.json
```
{
  "name": "Dashboard-Web",
  "version": "4.0.0",
  "scripts": {
    "tsc": "tsc",
    "tsc:w": "tsc -w",
    "ng": "ng",
    "start": "ng serve --ec true",
    "build": "ng build --prod",
    "test": "ng test",
    "lint": "ng lint"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.2.5",
    "@angular/common": "^5.2.5",
    "@angular/compiler": "^5.2.5",
    "@angular/core": "^5.2.5",
    "@angular/forms": "^5.2.5",
    "@angular/http": "^5.2.5",
    "@angular/platform-browser": "^5.2.5",
    "@angular/platform-browser-dynamic": "^5.2.5",
    "@angular/router": "^5.2.5",
    "@ng-bootstrap/ng-bootstrap": "^1.0.0",
    "@ngx-translate/core": "^9.1.1",
    "@ngx-translate/http-loader": "^2.0.1",
    "ajv": "^6.3.0",
    "core-js": "^2.5.3",
    "font-awesome": "^4.7.0",
    "highcharts": "^6.0.7",
    "highcharts-angular": "github:highcharts/highcharts-angular",
    "rxjs": "^5.5.6",
    "yarn": "^1.3.2",
    "zone.js": "^0.8.20"
  },
  "devDependencies": {
    "@angular/cli": "1.7.0",
    "@angular/compiler-cli": "^5.2.5",
    "@angular/language-service": "^5.2.5",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^9.4.7",
    "codelyzer": "~4.1.0",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "^1.4.1",
    "karma-jasmine": "~1.1.1",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.6.2"
  }
}
```

3. Updated tsconfig.json include with node_modules/highcharts-angular/src/app/* and Updated tsconfig.json files with the path to the ts file.  (TS was throwing an error when parse the include without the actually ts file being added to the config)
```
{
    "compileOnSave": true,
    "compilerOptions": {
        "outDir": "./dist/out-tsc.js",
        "sourceMap": true,
        "declaration": false,
        "moduleResolution": "node",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "lib": [
            "es2017",
            "dom"
        ],
        "listFiles": true,
        "baseUrl": "/dist",
        "paths": {
            "@app/*": ["dist/*"]
        },
        "pretty": true,
        "target": "es5",
        "typeRoots": [
         "node_modules/@types"
        ]
    },
    "files":[
        "node_modules/highcharts-angular/src/app/highcharts-chart.component.ts"
    ],
    "include":[
        "**/*.d.ts",
        "src/**/*",
        "node_modules/highcharts/*",
        "node_modules/highcharts-angular/src/app/*"
    ],
    "exclude": [
        "**/*.spec.ts"
    ]
}
```

4.  Added HighchartsChartComponent to chart.module.ts as a declaration.

```
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { ChartRoutingModule } from './chart-routing.module';
import { ChartComponent } from './chart.component';

import { HighchartsChartComponent } from 'highcharts-angular';

@NgModule({
    imports: [CommonModule, ChartRoutingModule],
    declarations: [ChartComponent, HighchartsChartComponent]
})
export class ChartModule {}
```

5.  Added the highcharts-chart directive to chart.component.html with [options]="chartOptions"

`<highcharts-chart [Highcharts]="Highcharts" [options]="chartOptions"></highcharts-chart>`

6.  Set chartOptions to simple highcharts json object.

```
import {Component, OnInit, Input, ViewChild, ElementRef} from '@angular/core';
import * as Highcharts from 'highcharts';

@Component({
  selector: 'app-chart',
  templateUrl: './chart.component.html',
  styleUrls: ['./chart.component.scss']
})
export class ChartComponent implements OnInit {
  @Input() title: any;
  chartOptions: any;
  Highcharts = Highcharts.Highcharts;

  constructor() {}

  ngOnInit() {
      this.showChart();
  }

  ngOnDestroy(){
  }


  showChart(){
      this.chartOptions = {
        chart: {
            type: 'line',
            height: '300px'
        },
        title: {text: ''},
        xAxis: {
            categories: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
        },
        yAxis: {
        },
        legend: {
            verticalAlign: 'bottom',
            layout: 'horizontal',
            align: 'center',
            itemStyle: {"cursor": "default"}
        },
        tooltip: {
            formatter: function () {
                return '<b>' + this.series.name + '</b><br/>' +
                    this.x + ': ' + this.y;
            }
        },
        plotOptions: {
        series: {
          animation: false,
          cursor: 'pointer'
        },
        bar: {
            animation: false,
            events: {
                legendItemClick: function () {
                    return false; // <== returning false will cancel the default action
                }
            }
        }
        },
        series: [{
            data: [29.9, 71.5, 106.4, 129.2, 144.0, 176.0, 135.6, 148.5, 216.4, 194.1, 95.6, 54.4]
        }]
      };
  }

}
```

7. Run ng build --watch ... no errors
8.  Load app in Chrome and receive this error:

> ERROR Error: Uncaught (in promise): Error: Template parse errors:
> Can't bind to 'Highcharts' since it isn't a known property of 'highcharts-chart'.
> 1. If 'highcharts-chart' is an Angular component and it has 'Highcharts' input, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
> 3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("
>     </h5>
>     
>     <highcharts-chart [ERROR ->][Highcharts]="Highcharts" [options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:22
> Can't bind to 'options' since it isn't a known property of 'highcharts-chart'.
> 1. If 'highcharts-chart' is an Angular component and it has 'options' input, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
> 3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. (">
>     
>     <highcharts-chart [Highcharts]="Highcharts" [ERROR ->][options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:48
> 'highcharts-chart' is not a known element:
> 1. If 'highcharts-chart' is an Angular component, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("
>     </h5>
>     
>     [ERROR ->]<highcharts-chart [Highcharts]="Highcharts" [options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:4
> Error: Template parse errors:
> Can't bind to 'Highcharts' since it isn't a known property of 'highcharts-chart'.
> 1. If 'highcharts-chart' is an Angular component and it has 'Highcharts' input, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
> 3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("
>     </h5>
>     
>     <highcharts-chart [ERROR ->][Highcharts]="Highcharts" [options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:22
> Can't bind to 'options' since it isn't a known property of 'highcharts-chart'.
> 1. If 'highcharts-chart' is an Angular component and it has 'options' input, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
> 3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. (">
>     
>     <highcharts-chart [Highcharts]="Highcharts" [ERROR ->][options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:48
> 'highcharts-chart' is not a known element:
> 1. If 'highcharts-chart' is an Angular component, then verify that it is part of this module.
> 2. If 'highcharts-chart' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("
>     </h5>
>     
>     [ERROR ->]<highcharts-chart [Highcharts]="Highcharts" [options]="chartOptions"></highcharts-chart>
> </div>
> "): ng:///DashboardModule/ChartComponent.html@6:4
>     at syntaxError (compiler.js:485)
>     at TemplateParser.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.TemplateParser.parse (compiler.js:24668)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._parseTemplate (compiler.js:34621)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileTemplate (compiler.js:34596)
>     at compiler.js:34497
>     at Set.forEach (<anonymous>)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileComponents (compiler.js:34497)
>     at compiler.js:34367
>     at Object.then (compiler.js:474)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileModuleAndComponents (compiler.js:34366)
>     at syntaxError (compiler.js:485)
>     at TemplateParser.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.TemplateParser.parse (compiler.js:24668)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._parseTemplate (compiler.js:34621)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileTemplate (compiler.js:34596)
>     at compiler.js:34497
>     at Set.forEach (<anonymous>)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileComponents (compiler.js:34497)
>     at compiler.js:34367
>     at Object.then (compiler.js:474)
>     at JitCompiler.webpackJsonp../node_modules/@angular/compiler/esm5/compiler.js.JitCompiler._compileModuleAndComponents (compiler.js:34366)
>     at resolvePromise (zone.js:809)
>     at resolvePromise (zone.js:775)
>     at zone.js:858
>     at ZoneDelegate.webpackJsonp../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
>     at Object.onInvokeTask (core.js:4740)
>     at ZoneDelegate.webpackJsonp../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
>     at Zone.webpackJsonp../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:188)
>     at drainMicroTaskQueue (zone.js:595)

**Comment by tleveque23** [comment]
I have the exact same problem. Why is this module not a real Angular package?
Please, if it is supposed to be the "official" module, it should be better than that.

The only way I am able to use it, is to copy the component inside my own source (and remove the dependencies to the module). It is working, but it's really really bad. If there is an update, I will have to copy it again manually!

**Comment by ChrystalLangston** [comment]
@tleveque23 Thank you!  What you've outlined worked for me and it actually surfaced an underlining issue with how our app's Angular routing set up.  We are using LoadChildren and the component was not associated with the module.ts defined under the path I was hitting.  Instead, the component was associated with a grandchild component.  Since it was the grandchild, it was never loading the module.ts and therefore it was throwing the error.  Once I added the component to the correct module.ts everything worked perfectly.  

I decided to revert back to the original set up, as outlined by highcharts-angular doc, just to see if that was the issue all along and guess what... everything loaded perfectly!  There is no longer a need for the ts file to be directly in our app.  I was not looking forward to having to copy the ts file over manually after every update either. :| I do hope that you will eventually be able to move away from that. Thank you again - your feedback helped immensely!!!

C

**Comment by tleveque23** [comment]
@ChrystalLangston I need to thank you too! Because after reading your last message, I was wondering why it is working for you and not me. Since I don't have the same lazy loading problem as you.
I looked more closely at your files I found that you also added this in your tsconfig file 
```
"files":[
        "node_modules/highcharts-angular/src/app/highcharts-chart.component.ts"
    ]
```
I gave it a try.... And it is now working!
That was the missing link for everything to work.
They need to add this I their documentation.
